### ISSUE TITLE
"PUT /api/v1/spool/45/use HTTP/1.1" 200 OK

### ISSUE THREAD

**Issue by mfraser** [Donkie/Spoolman#532]
I've been looking in the syslogs of my server trying to diagnose another issue when I noticed that when I'm printing, spoolman 0.19.3 is constantly writing to /var/log/syslog with INFO lines
```

Nov  9 08:50:24 server bash[110213]: INFO:     192.168.2.42:56306 - "PUT /api/v1/spool/45/use HTTP/1.1" 200 OK
Nov  9 08:50:29 server bash[110213]: INFO:     192.168.2.42:56314 - "PUT /api/v1/spool/45/use HTTP/1.1" 200 OK
Nov  9 08:50:34 server bash[110213]: INFO:     192.168.2.42:56316 - "PUT /api/v1/spool/45/use HTTP/1.1" 200 OK
Nov  9 08:50:39 server bash[110213]: INFO:     192.168.2.42:56324 - "PUT /api/v1/spool/45/use HTTP/1.1" 200 OK
Nov  9 08:50:44 server bash[110213]: INFO:     192.168.2.42:56326 - "PUT /api/v1/spool/45/use HTTP/1.1" 200 OK
Nov  9 08:50:49 server bash[110213]: INFO:     192.168.2.42:56340 - "PUT /api/v1/spool/45/use HTTP/1.1" 200 OK
```
Is there any way of turning these lines off?

**Comment by lattaai5** [comment]
Hello, I tried to fix the issue

This is what I did:

Modified the main.py file to add an environment variable SPOOLMAN_DISABLE_REQUEST_LOGGING that, when set to true, will disable console logging for requests. This should prevent the constant INFO lines from appearing in the syslog during printing operations.

I also created pull request: https://github.com/Donkie/Spoolman/pull/533


> [!CAUTION]
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!
>
> The fix provided by Latta AI might not be complete and it can serve as an inspiration.

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission

If you no longer want Latta AI to attempt fixing issues on your repository, you can block this account.

**Comment by Donkie** [comment]
You can change the logging level to warnings if you think the info level ones are superfluous, there is an environment variable for that.

**Comment by mfraser** [comment]
> You can change the logging level to warnings if you think the info level ones are superfluous, there is an environment variable for that.
Yes, I notice in spoolman/env.py this line:
`log_level_str = os.getenv("SPOOLMAN_LOGGING_LEVEL", "INFO").upper()`

Where is the best place to set SPOOLMAN_LOGGING_LEVEL to something else like ERROR or WARNING?

**Comment by mfraser** [comment]
> > You can change the logging level to warnings if you think the info level ones are superfluous, there is an environment variable for that.
> > Yes, I notice in spoolman/env.py this line:
> > `log_level_str = os.getenv("SPOOLMAN_LOGGING_LEVEL", "INFO").upper()`
> 
> Where is the best place to set SPOOLMAN_LOGGING_LEVEL to something else like ERROR or WARNING?

Found it in Spoolman/.env

```
# Logging level: DEBUG, INFO, WARNING, ERROR, CRITICAL
# Default if not set: INFO
#SPOOLMAN_LOGGING_LEVEL=INFO
```

**Comment by mfraser** [comment]
I've changed .env to
```
# Logging level: DEBUG, INFO, WARNING, ERROR, CRITICAL
# Default if not set: INFO
SPOOLMAN_LOGGING_LEVEL=ERROR
```
still getting INFO lines in my syslog.

**Comment by Donkie** [comment]
Found the issue here, Uvicorn did its own logging thing and didn't respect my log level.
